.github/workflows/lint.yaml

```diff
@@ -0,0 +1,70 @@
+name: Lint
+
+on:
+  pull_request:
+
+env:
+  GIT_USERNAME: ${{ github.actor }}
+  GIT_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+  FIX_COMMIT_MESSAGE: Fixed /hook_tools/lint.py lints
+
+jobs:
+  validate-content:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+          submodules: true
+      
+      - name: Set base commit (pull request)
+        run: |
+          echo "GITHUB_BASE_COMMIT=${{ github.event.pull_request.base.sha }}" >> $GITHUB_ENV
+          echo "GITHUB_CURRENT_COMMIT=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+        if: github.ref != 'refs/heads/main'
+
+      - name: Set base commit (main)
+        run: |
+          echo "GITHUB_BASE_COMMIT=${{ github.event.before }}" >> $GITHUB_ENV
+          echo "GITHUB_CURRENT_COMMIT=${{ github.event.after }}" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/main'
+      
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
```

Comment:
Can we use a more recent version of Python? As far as I know, versions 3.9 and below are no longer supported in the latest versions of Ubuntu.